### PR TITLE
To add Chained Scheduler to the list of PyTorch schedulers.

### LINKED ISSUE #4281: Update reference scripts to use the "Batteries Included" utils
## 🚀 Feature
As part of the "Batteries Included" initiative (#3911) we are adding a number of new utils that can be used to produce SOTA results. Once those utils are landed, we should update our reference scripts to use them. 

More specifically we need to:
- [ ] Update the reference scripts to use PyTorch's new [linear warmup](https://github.com/pytorch/pytorch/pull/60836) scheduler.
  - The reference scripts should be updated to let users define the `warmup_method` (if any), the `warmup_iters` and the `warmup_factor`. The warmup scheduler should be chained with other existing schedulers.
  - The Object Detection recipe needs to be BC compatible and replace our [custom linear warmup](https://github.com/pytorch/vision/blob/dc42f933f3343c76727dbfba6e4242f1bcb8e1a0/references/detection/engine.py#L20-L24) approach with the one from PyTorch.
  - Similarly the Video Classification recipe needs also to be BC compatible and replace our [customer scheduler](https://github.com/pytorch/vision/blob/master/references/video_classification/scheduler.py#L5).
  - All other recipes (Classification and Segmentation), should be updated to optionally use warm up. The addition should be BC compatible and turned off by default.
- [x] Update the Classification reference to use PyTorch's new label smoothing implementation.
  - We will support the case of label smoothing by using the upcoming `label_smoothing` argument of CrossEntropyLoss (https://github.com/pytorch/pytorch/pull/63122). 
  - A new `label_smoothing` argument should be introduced on the [script](https://github.com/pytorch/vision/blob/c2ab0c59f42babf9ad01aa616cd8a901daac86dd/references/classification/train.py#L165). To ensure BC, its default value should be 0.0 (no label smoothing).

### PR DESCRIPTION
**Pull request by iramazanli**
In this PR we are introducing ChainedScheduler which initially proposed in the discussion https://github.com/pytorch/pytorch/pull/26423#discussion_r329976246 .

The idea is to provide a user friendly chaining method for schedulers, especially for the cases many of them are involved and we want to have a clean and easy to read interface for schedulers. This method will be even more crucial once CompositeSchedulers and Schedulers for different type of parameters are involved.

The immediate application of Chained Scheduler is expected to happen in TorchVision Library to combine WarmUpLR and  MultiStepLR https://github.com/pytorch/vision/blob/master/references/video_classification/scheduler.py#L5 . However, it can be expected that in many other use cases also this method could be applied.

### Example
The usage is as simple as below:

```python
sched=ChainedScheduler([ExponentialLR(self.opt, gamma=0.9),
                        WarmUpLR(self.opt, warmup_factor=0.2, warmup_iters=4, warmup_method="constant"),
                        StepLR(self.opt, gamma=0.1, step_size=3)])
```

Then calling 
```python
sched.step()
```
would trigger step function for all three schedulers consecutively

Partially resolves https://github.com/pytorch/vision/issues/4281

### CODE REVIEW

File: torch/optim/lr_scheduler.py [hunk 1]
```diff
@@ -603,6 +603,44 @@ def _get_closed_form_lr(self):
                 for base_lr in self.base_lrs]
 
 
+class CombinedScheduler(object):
```
Comment (datumbox) [inline torch/optim/lr_scheduler.py:606]: Not sure if we should inherit from `_LRScheduler` but I think we need to implement `state_dict` and `load_state_dict` so that existing scripts will continue working without issues. See [here](https://github.com/pytorch/vision/blob/master/references/video_classification/train.py#L220) and [here](https://github.com/pytorch/vision/blob/master/references/video_classification/train.py#L239). I'm not sure if also implementing `get_last_lr`, `get_lr` and `print_lr` makes sense or not. Thoughts?

**Review by datumbox — COMMENTED** [review]
Thanks for the fast PR. I left a couple of comments, please let me know what you think.

File: torch/optim/lr_scheduler.py (hunk 1 above)
Comment (fmassa) [reply torch/optim/lr_scheduler.py:606]: I agree with @datumbox , I think making `CombinedScheduler` inherit from `_LRScheduler` would probably be more natural, as it effectively be a drop-in replacement for individual schedulers.
So `CombinedScheduler([scheduler])` would be semantically the same as `scheduler` for all operations that `scheduler` supports.

But this might come with some challenges,

**Review by datumbox — APPROVED** [review]
LGTM, thanks for your PR @iramazanli. 

I know that the name of the scheduler might change but other than that the code looks fine to me. Feel free to merge whenever you are happy.
